File: Easy/496-NextGreaterElementI.cpp
/*
   @author
   Akshat Atray
   Leetcode- https://leetcode.com/akshatatray/
   LeetCode Problem 496. Next Greater Element I
   Language C++
   Level Easy
*/

class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        stack<int> s;
        unordered_map<int, int> mp;
        vector<int> res;
        for (auto n : nums2) {
            while (s.size() && s.top() < n) {
                mp[s.top()] = n;
                s.pop();
            }
            s.push(n);
        }
        for (auto n : nums1)
            res.push_back(mp.count(n) ? mp[n] : -1);
        return res;
    }
};

File: Easy/496-NextGreaterElementI.py
'''
    @author
    Prajjwal Maurya
    Leetcode- https://leetcode.com/pmaurya6jun/
    Leetcode Problem 496.Next Greater Element I
    Language Python
    Level Easy
'''

class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res = []
        
        
        for num in nums1:
            flag = False
            for j in range(nums2.get(num)+1, len(nums2)):
                if nums2[j] > num:
                    res.append(nums2[j])
                    flag = True
                    break

            if flag == False:
                res.append(-1)
        return res